### PR TITLE
Move @fs/configurations webpack config into cra.

### PR DESCRIPTION
**Pull request by tylergraf**

### CODE REVIEW

File: packages/react-scripts/config/zion-webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const baseWebpack = require('./webpack.config.js')
+
+const craWebpack = baseWebpack('development')
+delete craWebpack.entry
+delete craWebpack.output
+delete craWebpack.plugins
+delete craWebpack.optimization
+
+module.exports = merge(craWebpack, {
+  externals: [nodeExternals(), 'react', 'react-dom', /^@fs/, /^@fs-sandbox/],
```
Comment (joeycozza) [inline packages/react-scripts/config/zion-webpack.config.js]: didn't we have prop-types in the explicit externals array here? and @emotion/core?

Comment (tylergraf) [reply packages/react-scripts/config/zion-webpack.config.js]: I don't know. I copied this from https://github.com/fs-webdev/configurations/blob/master/webpack.config.js#L16

Comment (joeycozza) [reply packages/react-scripts/config/zion-webpack.config.js]: oh yeah. I added those 2 specifically inside of zion.

File: packages/react-scripts/proxy/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        devServer: {
+          // setupProxy optionally takes customProxies as 2nd arg, if needed
+          before: app => setupProxy(app),
+        },
+      },
+    ],
+    // to here
+    '@neutrinojs/airbnb',
+    '@neutrinojs/react-components',
+    '@neutrinojs/jest',
```
Comment (redbugz) [inline packages/react-scripts/proxy/README.md]: do we need neutrino anymore, now that this is out of exo?

File: packages/react-scripts/proxy/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+```sh
+REMOTE_ENV=int
+```
+
+- Apps can set their own local proxy overrides
+- Config options from http-proxy-middleware
+- [https://github.com/chimurai/http-proxy-middleware#http-proxy-options](https://github.com/chimurai/http-proxy-middleware#http-proxy-options)
+- If others will use the service you add, please add it to the default proxy config here:
+- [https://github.com/fs-webdev/exo/blob/master/proxy/proxies.js](https://github.com/fs-webdev/exo/blob/master/proxy/proxies.js)
```
Comment (redbugz) [inline packages/react-scripts/proxy/README.md]: fix this link?

File: packages/react-scripts/scripts/utils/frontierInit.js [hunk 4]
```diff
@@ -68,12 +67,12 @@ function installFrontierDependencies(appPath, appName, answers, ownPath) {
     ...[
       '@fs/eslint-config-frontier-react',
       '@fs/zion-testing-library',
+      '@fs/exo',
```
Comment (redbugz) [inline packages/react-scripts/scripts/utils/frontierInit.js]: do we need exo still?

File: packages/react-scripts/config/zion-webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  resolve: {
+    alias: {
+      // fix duplicate copies of react issue for hooks
+      // https://github.com/webpack/webpack/issues/8607
+      react: require.resolve('react'),
+      'react-dom': require.resolve('react-dom'),
+    },
+  },
+  // TODO: we can probably remove the whole module prop once babel-preset-frontier gets into react-scripts
```
Comment (redbugz) [inline packages/react-scripts/config/zion-webpack.config.js]: has this happened yet?

File: packages/react-scripts/package.json [hunk 6]
```diff
@@ -52,8 +56,10 @@
     "file-loader": "2.0.0",
     "fork-ts-checker-webpack-plugin-alt": "0.4.14",
     "fs-cli-goodies": "github:fs-webdev/fs-cli-goodies",
+    "fs-config": "github:fs-webdev/fs-config",
```
Comment (redbugz) [inline packages/react-scripts/package.json]: is this going to be hard to keep up to date in here?

File: packages/react-scripts/proxy/README.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
+## Proxy
+
+- Use this proxy in localdev to talk to your environment's APIs, control their timeouts, etc.
+- Requires Node 10+ during localdev.
+- App must run on PORT 5000-5009 to comply with auth service whitelist, until we can get the default PORT 3000 added to it
+
+### Usage
+
+- In your app (if not installed by create-react-app for you already):  
+  `npm install --save-dev fs-webdev/exo`
```
Comment (redbugz) [inline packages/react-scripts/proxy/README.md]: this readme needs several updates I think

File: packages/react-scripts/proxy/README.md (hunk 2 above)
Comment (tylergraf) [reply packages/react-scripts/proxy/README.md]: Nope

File: packages/react-scripts/scripts/utils/frontierInit.js (hunk 4 above)
Comment (tylergraf) [reply packages/react-scripts/scripts/utils/frontierInit.js]: no

File: packages/react-scripts/config/zion-webpack.config.js (hunk 5 above)
Comment (tylergraf) [reply packages/react-scripts/config/zion-webpack.config.js]: I thought it had, but when I remove this, it fails to transpile.

Comment (redbugz) [reply packages/react-scripts/config/zion-webpack.config.js]: @joeycozza do you know where we are on this?

Comment (joeycozza) [reply packages/react-scripts/config/zion-webpack.config.js]: Exactly where Tyler is. The code in react-scripts/config/webpack.config.js does have a rule using @fs/babel-preset-frontier, BUT there are more configurations/options set in that one than there are here in this zion-webpack.config.

So I assume that some of those options aren't good for what zion wants/needs.

Comment (redbugz) [reply packages/react-scripts/config/zion-webpack.config.js]: So do we remove the TODO? Any idea who would know the final status on this?

Comment (joeycozza) [reply packages/react-scripts/config/zion-webpack.config.js]: I don't think there is a final status. We could either do some webpack and babel deep dive to see why removing this here breaks zion, or we could just remove the TODO and let this be.

Comment (tylergraf) [reply packages/react-scripts/config/zion-webpack.config.js]: I think we should dive into it when we can. I'll create a card.

**Comment by tylergraf** [comment]
@redbugz I think this is ready.

**Comment by tylergraf** [comment]
@joeycozza what do you think?

**Pull request merged**